core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -1935,22 +1929,19 @@ private long loadDirectoryMetadataAndJournal(LockedInodePath inodePath,
   }
 
   /**
-   * Loads metadata for the path if it is (non-existing || (persisted && load direct children is
-   * set).
+   * Loads metadata for the path if it is (non-existing || load direct children is set).
    *
    * @param inodePath the {@link LockedInodePath} to load the metadata for
    * @param options the load metadata options
    */
-  // TODO(peis): Add a unit test for this function.
   private long loadMetadataIfNotExistAndJournal(LockedInodePath inodePath,
       LoadMetadataOptions options) {
     boolean inodeExists = inodePath.fullPathExists();
     boolean loadDirectChildren = false;
     if (inodeExists) {
       try {
         Inode<?> inode = inodePath.getInode();
-        loadDirectChildren =
-            inode.isDirectory() && inode.isPersisted() && options.isLoadDirectChildren();
+        loadDirectChildren = inode.isDirectory() && options.isLoadDirectChildren();
```

Comment:
Oh okay, could you add a unit test to verify the unpersisted dir gets marked persisted if we end up loading something from the ufs? Also, how will this work if the unpersisted files overlap with the persisted file, or the security bits of the unpersisted directory conflict with the persisted directory?